tt_eager/tt_numpy/functions.hpp

```diff
@@ -189,6 +189,85 @@ static Tensor index_trilu(const Shape& shape, const int32_t diag, DataType data_
     return output;
 }
 
+
+template<typename T>
```

Comment:
Assert for rank 4 in these functions since your loops are hardcoded.

TODO is to remove the hardcoded dim loops to generically support any dim and possibly refactor into single function (not a requirement for this pr).